src/int/mul.rs

```diff
@@ -0,0 +1,15 @@
+//! [`Int`] multiplication operations.
+
+use subtle::{ConstantTimeEq, CtOption};
+
+use crate::int::Int;
+use crate::{CheckedMul, Uint, Zero};
+
+impl<const LIMBS: usize, const RHS_LIMBS: usize> CheckedMul<Int<RHS_LIMBS>> for Int<LIMBS> {
+    #[inline]
+    fn checked_mul(&self, rhs: &Int<RHS_LIMBS>) -> CtOption<Self> {
```

Comment:
same as for div, since there are two zero representation the sign should be preserved.
i.e. 
* -0 * -0 = +0
* +0 * +0 = +0
* -0 * +0 = -0
* +0 * -0 = -0